streaming/base/storage/download.py

```diff
@@ -190,6 +208,22 @@ def download_from_gcs(remote: str, local: str) -> None:
         raise
 
 
+def _gcs_with_service_account(local: str, obj: urllib.parse.ParseResult) -> None:
+    """Download a file from remote GCS to local using service account credentials.
+
+    Args:
+        local (str): Local path (local filesystem).
+        obj (ParseResult): ParseResult object of remote path (GCS).
+    """
+    from google.cloud.storage import Blob, Bucket, Client
+
+    service_account_path = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+    gcs_client = Client.from_service_account_json(service_account_path)
+
+    blob = Blob(obj.path.lstrip('/'), Bucket(gcs_client, obj.netloc))
+    blob.download_to_filename(local)
```

Comment:
Can you check the in-progress filename by google service account ? For example, if rank 0 is downloading an `index.json` file with let say 64MB and other ranks wait for file `index.json` to exists, then, if GCS downloads it as `index.json` with split chunks, meaning, the file is not downloaded fully and other ranks thinks that it is downloaded fully and starts loading it and this would cause JSONDecoder error. However, if GCS downloads it as some temp extension such as `index.json.tmp` or something similar and then later renames it to `index.json` then we are good here.

Reply:
It seems like it opens a file object which it writes into but the file for a file object doesn't show up in the file system until after it's done writing.
https://github.com/googleapis/python-storage/blob/main/google/cloud/storage/blob.py#L1255